training/bing_bert/export_from_tensorboard.py

```diff
@@ -1,38 +1,47 @@
+#!/usr/bin/env python3
+
 from tbparse import SummaryReader
 import pandas as pd
 import argparse
 
-
+RUNTIME_NAME = 'Runtime (sec)'
 
 def parse_tb2(logdir, csv_out):
-	reader = SummaryReader(logdir, extra_columns=set(['wall_time']))
-	df = reader.scalars
-
-	df['wall_clock'] = pd.to_datetime(df.wall_time, unit="s")
-	first = df.wall_time[0]
-	df['Runtime'] = df.wall_time - first
-	
-	loss_df = df[df['tag'] == 'Train/Samples/train_loss'].copy()
-	loss_df.rename(columns = {'step' : 'Samples', 'value' : 'Loss'}, inplace=True)
-	loss_df.drop(columns = ['tag', 'wall_time', 'wall_clock'], inplace=True)
-	
-	
-	if csv_out:
-#		df[df['tag'] == 'Train/Samples/train_loss'].to_csv(csv_out)
-		loss_df.to_csv(csv_out)
-	
-	return loss_df
-
-	
-	
-	
-	
+    reader = SummaryReader(logdir, extra_columns=set(['wall_time']))
+    df = reader.scalars
 
+    loss_df = df[df['tag'] == 'Train/Samples/train_loss'].reset_index(drop=True)
 
-if __name__ == '__main__':
-	parser = argparse.ArgumentParser(description='Parse TensorBoard Event files and prints them')
-	parser.add_argument('logdir', help='TensorBoard logdir')
-	parser.add_argument('--csv_out', help='csv output file', required=False, default=None)
-	args = parser.parse_args()
+    loss_df['wall_clock'] = pd.to_datetime(loss_df.wall_time, unit="s")
+    first = loss_df.wall_time.iloc[0]
+    loss_df[RUNTIME_NAME] = loss_df.wall_time - first
+
+    loss_df.rename(columns={'step': 'Samples', 'value': 'Loss'}, inplace=True)
+    loss_df.drop(columns=['tag', 'wall_time', 'wall_clock'], inplace=True)
+
+    if csv_out:
+        loss_df.to_csv(csv_out)
 
-	print(parse_tb2(args.logdir, args.csv_out))
+    return loss_df
+
+
+def find_runtime(df, min_iters, loss_limit):
+    for d in df.rolling(min_iters, method="table"):
+        if d.shape[0] == min_iters and (d['Loss'] < loss_limit).all():
```

Comment:
It creates a rolling window of `min_iters` (=2) rows and checks if all losses are under the threshold. If they are it returns the time of the last row in the current window.

Example:
```
Loss  Time
12        1
11        2
10        3
09        4
08        5
07        6
06        7
```

For a threshold of 9.5 it would return 5 as the following window matches the condition:
```
09        4
08        5
```

Reply:
That depends on how the requirement is defined. Currently "runtime required to reach a train loss of 8.5 for at least 2 iterations" which I understand as "2 iterations with that loss must have passed". But I guess this is cleared up now as you used the same in cc5995090bd335b4d5535a943831e2c224081972